internal/build/job/signer.go

```diff
@@ -0,0 +1,175 @@
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+////go:generate mockgen -source=signer.go -package=signer -destination=mock_signer.go
+
+type signer struct {
+	name   string
+	scheme *runtime.Scheme
+}
+
+func NewSigner(scheme *runtime.Scheme) Maker {
+	return &signer{name: "sign", scheme: scheme}
+}
+
+func (m *signer) GetName() string {
+	return m.name
+}
+
+func (m *signer) ShouldRun(mod *kmmv1beta1.Module, km *kmmv1beta1.KernelMapping) bool {
+	if (mod.Spec.ModuleLoader.Container.Build == nil || mod.Spec.ModuleLoader.Container.Build.Sign == nil) && (km.Build == nil || km.Build.Sign == nil) {
+		return false
+	}
+	return true
+}
+
+func (m *signer) MakeJob(mod kmmv1beta1.Module, buildConfig *kmmv1beta1.Build, targetKernel string, previousImage string, containerImage string, pushImage bool) (*batchv1.Job, error) {
+	var args []string
+
+	if pushImage {
+		args = []string{"-signedimage", containerImage}
+	} else {
+		args = append(args, "-no-push")
+	}
+
+	if previousImage != "" {
+		args = append(args, "-unsignedimage", previousImage)
+	} else if buildConfig.Sign.UnsignedImage != "" {
+		args = append(args, "-unsignedimage", buildConfig.Sign.UnsignedImage)
+	} else {
+		return nil, fmt.Errorf("no image to sign given")
+	}
+	args = append(args, "-key", "/signingkey/key.priv")
+	args = append(args, "-cert", "/signingcert/public.der")
+
+	if len(buildConfig.Sign.FileList) > 0 {
+		args = append(args, "-filestosign", strings.Join(buildConfig.Sign.FileList, ":"))
+	}
+
+	fmt.Printf("buildConfig == %#v\n", buildConfig)
+	fmt.Printf("buildConfig.Sign == %#v\n", buildConfig.Sign)
+
+	volumes := []v1.Volume{}
+	volumeMounts := []v1.VolumeMount{}
+	volumes = append(volumes, m.makeImageSigningSecretVolume(buildConfig.Sign.KeySecret, "key", "key.priv"))
+	volumes = append(volumes, m.makeImageSigningSecretVolume(buildConfig.Sign.CertSecret, "cert", "public.der"))
+
+	volumeMounts = append(volumeMounts, m.makeImageSigningSecretVolumeMount(buildConfig.Sign.CertSecret, "/signingcert"))
+	volumeMounts = append(volumeMounts, m.makeImageSigningSecretVolumeMount(buildConfig.Sign.KeySecret, "/signingkey"))
+
+	if mod.Spec.ImageRepoSecret != nil {
+		args = append(args, "-pullsecret", "/docker_config/config.json")
+		volumes = append(volumes, m.makeImagePullSecretVolume(mod.Spec.ImageRepoSecret))
+		volumeMounts = append(volumeMounts, m.makeImagePullSecretVolumeMount(mod.Spec.ImageRepoSecret))
+	}
+
+	job := &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: mod.Name + "-" + m.GetName() + "-",
+			Namespace:    mod.Namespace,
+			Labels:       labels(mod, targetKernel, m.GetName()),
+		},
+		Spec: batchv1.JobSpec{
+			Completions: pointer.Int32(1),
+			Template: v1.PodTemplateSpec{
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{
+						{
+							Name:         "signimage",
+							Image:        "quay.io/chrisp262/kmod-signer:latest",
```

Comment:
Could it become a setting rather than a hard coded value in the Go code?
Shouldn't it point to an image released under the KMM project? And if yes, could you provide the Dockerfile in the PR to make it complete?

Reply:
> Could it become a setting rather than a hard coded value in the Go code?

Is that to be able to run substitutions when creating the bundle?
Besides that, I hardly see a use case for users to provide their own signer.

Reply:
the source for that image comes from https://github.com/chr15p/kmod_signer  and the idea is to move that code/Dockerfile into the kmm project as a separate PR, at which point this would obviously get updated from pointing at my quay.io repo.
I did consider making it user configurable but as this points to the only image in the world that provides this functionality it didn't seem like a high priority :)